doc/pages/faq.asciidoc

```diff
@@ -101,97 +206,117 @@ However, you can easily declare key mappings in your configuration file
+Also note that a subset of "readline shortcuts" is implemented for command
+prompts.
 
-The server expands shell scopes using the `sh` binary, stored in one of the
-directories where all the POSIX standard utilities can be found -this list
-of directories is stored in a system configuration variable, and queried
-by Kakoune at startup.
-
-In most distributions, `/bin/sh` will end up being used.
-
-== Can I disable auto-indentation completely ?
+=== Can I disable auto-indentation completely?
 
 All the indentation hooks are conventionally named `<lang>-indent`, which
 allows us to use the `disabled_hooks` variable to disable indentation
 globally with the following command: `set global disabled_hooks '.+-indent'`
 
-== How to enable syntax highlighting ?
+=== How to enable syntax highlighting?
 
-The mimetype of the files opened in new buffers is detected using the
+The MIME type of the files opened in new buffers is detected using the
 `file` command, and syntax highlighting enabled automatically when
 possible.
 
-== My file seems to be highlighted with the wrong colors, I thought syntax highlighting was detected automatically ?
+=== My file seems to be highlighted with the wrong colors, I thought syntax highlighting was detected automatically?
 
 The `file` utility has several shortcomings, such as detecting the
-wrong mimetype for a file containing data with different syntax, e.g.
+wrong MIME type for a file containing data with different syntax, e.g.
 a Python script containing hardcoded HTML templates detected as an HTML
 file.
 
-Kakoune does its best at detecting file types (using known extensions
-for a given format for instance), but not much can be done about those
-ambiguous cases. You might consider writing a custom `$HOME/.magic` file
-if needed.
+Kakoune does its best to detect file types (using known extensions for a
+given format for instance), but not much can be done about those ambiguous
+cases. You might consider writing a custom `$HOME/.magic` file if needed.
 
-== Can I disable syntax highlighting completely ?
+=== Can I disable syntax highlighting completely?
 
 Similarly to the indentation hooks, the name format followed by the
 highlighting hooks is `<lang>-highlight`. You can thus disable syntax
 highlighting using the following command: `set global disabled_hooks
 '.+-highlight'`
 
-== Why does a dot `.` in a regex select newline characters ?
+=== Can the cursor be rendered as a beam?
+
+Rendering the cursor as a beam is a common feature of other modal editors,
+it however doesn't fit within Kakoune's selection-first paradigm.
+
+There is a selection on screen at all times, containing either data selected
+by the user, or a newline character when the buffer is empty.
+
+A selection is bound by an anchor and a cursor. They can overlap, but
+ultimately must both be placed *over* a character. A beam cursor placed
+*between* two characters doesn't fulfil that requirement, and is thus
+not allowed.
```

Comment:
I think this is backwards: it's not that the cursor must be a block because a selection is minimum-length 1; selections are minimum-length 1 because the cursor must be a block.

The real reason the cursor must be a block is that terminals can only draw one "real" cursor (beam or otherwise) at a time, so when Kakoune has multiple selections, it has to draw them itself by setting foreground/background colours. Because a custom background colour must cover at least one character, therefore the Kakoune selection model has minimum-length 1.

Reply:
There always has to be at least one character selected, not because of a technical detail related to colours or the terminal, but because of the selection-first paradigm.

0-length selections simply don't make sense within the editing language.